Framework/Algorithms/src/CalculateTransmissionBeamSpreader.cpp

```diff
@@ -289,9 +289,9 @@ CalculateTransmissionBeamSpreader::fitToData(API::MatrixWorkspace_sptr WS) {
     b = std::pow(10, b);
     m = std::pow(10, m);
 
-    const MantidVec &X = result->readX(0);
-    MantidVec &Y = result->dataY(0);
-    MantidVec &E = result->dataE(0);
+    auto &X = result->x(0);
+    auto &Y = result->mutableY(0);
+    auto &E = result->mutableE(0);
     for (size_t i = 0; i < Y.size(); ++i) {
       Y[i] = b * (std::pow(m, 0.5 * (X[i] + X[i + 1])));
```

Comment:
This is computing bin centers, use `points()` instead of `x()`. Also add this to our list of potentially required math operations with histograms.